### DISCUSSION TITLE
support tables holding multiple entity types

### DISCUSSION THREAD

**Discussion by marcostorto** [directus/directus#15682] [Archive]
Is there a way to support a table holding multiple entity types, based on a discriminator column? I would like to have each entity type show up as a separate collection in directus

**Comment by br41nslug** [comment]
What you're describing sounds like what the "Many-to-any" interface does which is essentially a M2M relation with a discriminator column deciding which collection it is linking to.
https://docs.directus.io/configuration/data-model/relationships.html#many-to-any-m2a

**Reply by marcostorto** [reply]
I think it does not match my use case: I have an existing database table (equipments) which contains multiple different equipment types, with one column (type) which is used to discriminate the type per each row. 
I would like to show it in directus as separate collections (one per type) rather than as a single collection
Is this possible?

**Reply by br41nslug** [reply]
What kind of database are you using?

> with one column (type) which is used to discriminate the type per each row.

This does not fit with a relational SQL databases

**Reply by marcostorto** [reply]
We're using postgresql
Actually this use (multiple entities children of a common parent saved in a single database table mapping the common parent) is pretty common in Java applications (e.g. [https://www.javatpoint.com/jpa-single-table-strategy](https://www.javatpoint.com/jpa-single-table-strategy))
It allows to keep common attributes in the parent entity and manage the entities both as a single collection of the common parent type and as a separate collections of the child types depending on the operation required.

**Reply by br41nslug** [reply]
Ah i think i was assuming you meant entity on the database level 🤔 There is no problem with storing such a structure but i don't see Directus integrating well with it because it will mirror the direct database structure without abstraction so you'd only see the parent as a collection. You could possibly implement this in a custom module or on the frontend but there is no tooling built-in for this usecase
